docs/integration/webapi.rst

```diff
@@ -154,79 +163,136 @@ Instead of deriving from one of the existing Web API filter attributes your clas
+Note in the sample there's no actual async code that runs so it returns ``Task.FromResult(0)``, which is a common way to return an "empty task." 
+If your filter does require async code, you can return a real ``Task`` object or use ``async``/``await`` code just like any other asynchronous method.
+
+Continuation Action Filter Interface
+*************************************
+
+In addition to the regular ``IAutofacActionFilter`` in the example above, there is also an ``IAutofacContinuationActionFilter``. This interface
+also functions as an Action Filter, but rather than having an ``OnActionExecutingAsync`` and ``OnActionExecutedAsync`` method, it follows the continuation
+style with a single ``ExecuteActionFilterAsync`` method that takes a callback to run the next filter in the chain.
+
+You may want to use ``IAutofacContinuationActionFilter`` instead of ``IAutofacActionFilter`` if you want to wrap the entire request in a ``using`` block,
+such as when you want to allocate a ``TransactionScope`` to the request, like so:
+
+.. sourcecode:: csharp
+
+    public class TransactionScopeFilter : IAutofacContinuationActionFilter
+    {
+        public async Task<HttpResponseMessage> ExecuteActionFilterAsync(
+            HttpActionContext actionContext,
+            CancellationToken cancellationToken,
+            Func<Task<HttpResponseMessage>> next)
+        {
+            using (new TransactionScope(TransactionScopeAsyncFlowOption.Enabled))
+            {
+                return await next();
+            }
+        }
+    }
+
+.. note:: 
+
+  The regular ``IAutofacActionFilter`` runs inside a continuation filter, so async context is also preserved there between
+  ``OnActionExecutingAsync``, the action method itself, and the ``OnActionExecutedAsync`` of the filter. 
 
 Register the Filter
 -------------------
 
-For the filter to execute you need to register it with the container and inform it which controller, and optionally action, should be targeted. This is done using the following ``ContainerBuilder`` extension methods:
+For the filter to execute you need to register it with the container and inform it which controller (or controllers), and optionally action, should be targeted. 
+This is done using the following ``ContainerBuilder`` extension methods, which exist for each filter type:
 
-- ``AsWebApiActionFilterFor<TController>()``
-- ``AsWebApiActionFilterOverrideFor<TController>()``
-- ``AsWebApiAuthorizationFilterFor<TController>()``
-- ``AsWebApiAuthorizationOverrideFilterFor<TController>()``
-- ``AsWebApiAuthenticationFilterFor<TController>()``
-- ``AsWebApiAuthenticationOverrideFilterFor<TController>()``
-- ``AsWebApiExceptionFilterFor<TController>()``
-- ``AsWebApiExceptionOverrideFilterFor<TController>()``
+``AsWebApi{FilterType}ForAllControllers``
+  Register this filter to run for all action methods on all controllers, in the same way as registering a global Web API filter.
```

Comment:
This is a good refactoring of the doc, but on initial read it's not clear what options there are for `{FilterType}`. I do see the list down below, but maybe it would be more clear to move it up here where I see the first occurrence of `{FilterType}`; or put something in here that says "you can see the list of supported filter types below." Something to help folks connect the stuff up here to the stuff down there.

Reply:
I've moved the filter types up to the top of the section; I expect users don't want to have to scroll down first, then scroll up again.